tests/manual/test-procedure.rst

```diff
@@ -0,0 +1,282 @@
+Manual Test Procedure
+=====================
+
+Intro
+-----
+
+Until we have a comprehensive set end-to-end tests, we will have to carry
+some tests manually to ensure all the user-facing features of the Python Editor
+are working as expected.
+
+We need to be pragmatic about the amount of tests we carry manually, so this
+Test Procedure tries to balance coverage vs test duration.
+
+
+Nomenclature and Definitions
+----------------------------
+
+- **Python Editor**: This constitutes the whole web app.
+
+- **Text Editor**: The text or code editor (using the Ace framework) inside the
+  Python Editor.
+
+- **Load by editor drag&drop**: Load a file into the Python Editor by dragging
+  a file from the user OS, to the Text Editor area.
+
+- **Load by drop area drag&drop**: Load a file into the Python Editor by
+  clicking on the "Load button" and then dragging a file from the user OS to
+  the large grey box that says "Drag and drop a .hex or .py file in here to
+  open it".
+
+- **Load by file picker**: Load a file into the Python Editor by clicking on
+  the "Load button", then clicking on the "Or pick a file" link, then clicking
+  on the "Browse" button, selecting a file in the browser file picker, and
+  finally clicking on the "Load" button.
+
+- **Load by any method**: Load a file into the Python Editor using either
+  of the three options: "Load by editor drag&drop", "Load by drop area
+  drag&drop", or "Load by file picker".
+
+- **Connect to the REPL**: Use any type of serial terminal to connect with the
+  micro:bit, and send a Ctrl+C command to enter the REPL.
+
+- **Script Name**: The name of the script stored on the top right side of the
+  Python Editor. This can be often confused with a search box.
+
+- **Test Case**: A set of test steps to verify a related set of conditions.
+
+
+Preparation
+-----------
+
+Ensure you have everything listed in the "Prerequisites" section.
+
+Each Test Case will focus on a single feature under tests. Make sure to read
+the "Test Cases" opening paragraphs carefully.
+
+
+You will need
+'''''''''''''
+
+- Internet Explorer 10
+
+- A modern version of Edge
+
+- A modern version of Chrome
+
+- A modern version of Firefox
+
+- A text editor (the one included in most operating systems is fine).
+
+- Access to the Python Editor under test
+
+- The test files from
+  https://github.com/bbcmicrobit/PythonEditor/tree/master/tests/manual/test-files
+
+- A micro:bit
+
+- One from the following:
+
+  - A serial terminal program you are comfortable to connect to a micro:bit
+  - The working microREPL installation (https://github.com/ntoll/microrepl)
+  - Mu editor (https://codewith.mu/)
+  - A micro:bit with WebUSB-enabled DAPLink and access to a version of the
```

Comment:
Should we link to firmware update page here https://microbit.org/guide/firmware/ and say this is 0247+

Reply:
As with the previous one, I'd prefer to assume the person running the tests can figure this out themselves. This part is not essential either, as you only need one option from the list.